src/content_delegate.js

```diff
@@ -585,31 +588,193 @@ ContentDelegate.prototype.attachRecapLinkToEligibleDocs = function() {
+        return new Blob([buffer], {type: 'application/zip'});
+      })
+    const blobUrl = URL.createObjectURL(zipFile);
+    const pacerCaseId = (event.data.id).match(/caseid\=\d*/)[0].replace(/caseid\=/, "");
+
+    // load options
+    const payload = await getItemsFromStorage(['options']);
+    const options = payload['options'];
+
+    if (options['recap_enabled'] && !this.restricted) {
+      this.recap.uploadZipFile(
+        this.court, // string
+        pacerCaseId, // string
+        nonce, // string
+        (ok) => { // callback
+          if (ok) {
+            const filename = generateFileName(options, pacerCaseId);
+            // convert htmlPage to document
+            const link =
+              `<a id="recap-download" href=${blobUrl} download=${filename} width="0" height="0"/>`;
+            const htmlBody = stringToDocBody(htmlPage);
+            const frame = htmlBody.querySelector('iframe');
+            frame.insertAdjacentHTML('beforebegin', link);
+            frame.src = "";
+            frame.onload = () => document.getElementById('recap-download').click();
+            document.body = htmlBody;
+            history.pushState({content: document.body.innerHTML}, '');
+            // show notifier
+            this.notifier.showUpload('Zip uploaded to the Public Recap Archive', () => {});
+            saveItemToStorage({[nonce]: ""});
+            saveItemToStorage({'docId': ""});
+          }
+        }
+      );
     }
-  }, this));
+  } catch (err) {
+    console.error(err);
+  }
 };
+
+// Same as handlePDFView, but for zip files!
+ContentDelegate.prototype.handleZipFilePageView = function () {
+  // return if not the download all page
+  if (!PACER.isDownloadAllDocumentsPage(this.url, document)) {
+    return;
+  }
+
+  // extract the url from the onclick attribute from one of the two
+  // "Download Documents" buttons
+  const inputs = [...document.getElementsByTagName("input")];
+  const targetInput = inputs.find(
+    input => input.type === "button" && input.value === "Download Documents"
+  );
+  const url = targetInput
+    .getAttribute("onclick")
+    .replace(/p.*\//, "") // remove parent.location='/cgi-bin/
+    .replace(/\'(?=$)/, ""); // remove endquote
+
+  // now we replace the onclick method of the "Download Documents" buttons
+  // with the postMessage function and zero out the form action buttons
+  // prettier-ignore
+  const dangerouslySetInnerHTML = [
+    'let forms = document.forms;',
+    'for (i = 0; i < forms.length; i++) {',
+    'let form = forms[i];',
+    'form.removeAttribute("action")',
+    '}',
+    'let items = document.getElementsByTagName("input");',
+    'for (i = 0; i < items.length; i++) {',
+    'input = items[i];',
+    'if (input.type === "button" && input.value === "Download Documents") {',
+    'input.removeAttribute("onclick");',
+    `input.addEventListener("click", () => window.postMessage({ id: ${JSON.stringify(url)}}))`,
+    '};',
+    '}'
+  ].join('');
```

Comment:
This is in lieu of the monkey patch approach for PDFs? Can you make a comment about how it's dangerous, what bad things could be done or happen, etc? (Assume I'm bad at this.)

Reply:
Narrator: He **was** pretty bad at it.

Reply:
I mean the danger is pretty self evident, no? Injecting html means arbitrary code execution ... which is dangerous. I'll put a comment beforehand, but not sure how detailed you want me to be.

Reply:
I mean, yes, injecting HTML/JS is bad, but there are no variables here, right, so I'm not sure I understand what an abuse/attack of this would look like.

Reply:
You’re giving a program the ability to write js that will execute immediately on the browser. That JS could access data on the page, potentially account credentials, or it could install or run malware, or hijack your browser to mine bitcoin ... hence why injecting JS as html is dangerously set